### ISSUE TITLE
How to upgrade PostgreSQL version (currently at 9.1)?

### ISSUE THREAD

**Issue by vlucas** [dokku-alt/dokku-alt#234]
I am loving the simplicity of using dokku and dokku-alt, so huge thanks for this project!

I ran into a blocking issue on my project that requires the JSON PostgreSQL data type. I dug around a bit, and discovered that by default, dokku-alt appears to use PostgreSQL 9.1 (which doesn't support json).

See which version is currently running:
```
dokku postgresql:console <app> <db>
SELECT version();
```

I am new to docker and the whole container approach. How do I go about upgrading to PG 9.3 or 9.4? I noticed you already have branches to support this: https://github.com/dokku-alt/postgresql-dockerfiles/tree/9.4 - any tips or steps would be greatly appreciated. Thanks!

**Comment by jerryasmith** [comment]
Looking for help with a solution here as well.

**Comment by ayufan** [comment]
The only option is to export databases and reimport them after image change.

**Comment by vlucas** [comment]
Sure - what commands need to be run to acheive that?

Also - I think the main installation choses PG 9.1 because `9.1` is set as the default branch in the repo: https://github.com/dokku-alt/postgresql-dockerfiles - the solution might be as simple as setting 9.3 or 9.4 to the default branch and then I can re-install dokku-alt.

**Comment by ayufan** [comment]
Yes, but it's not that simple, because database needs to be migrated to newer version which is not that strightforward.

**Comment by vlucas** [comment]
I understand what you're saying, but I literally just setup dokku-alt and have no data in PostgreSQL at all. I stopped as soon as I realized it was only PG 9.1 and filed this ticket. I could completely destroy the VPS and create a new one and would lose nothing. So for me right now with no data in it at all, it is indeed that simple. I just want to know how to install dokku-alt from a fresh install with PG 9.4. That's all I am asking. By default, it uses 9.1 right now.

**Comment by ayufan** [comment]
Ok. You have to add to /home/dokku/.dokkurc:
```bash
export POSTGRESQL_IMAGE=ayufan/dokku-alt-postgresql:9.3
```

(The 9.3 is latest provided)

and later execute:
```bash
rm -rf /home/dokku/.pgsql
dokku postgresql:restart
```

**Comment by georgewhewell** [comment]
@ayufan This doesn't work for me :(

    $ dpkg -s dokku-alt-beta | grep Version
    Version: 0.3.10-146-g1ee5f7c

    $ cat /home/dokku/.dokkurc
    export POSTGRESQL_IMAGE=ayufan/dokku-alt-postgresql:9.3

    $ dokku postgresql:restart
    -----> Stopping PostgreSQL container...
    -----> Starting PostgreSQL container...
    -----> PostgreSQL container started!

    $ dokku postgresql:console:admin
    psql (9.1.14)
    SSL connection (cipher: DHE-RSA-AES256-SHA, bits: 256)
    Type "help" for help.

    root=#

**Comment by vlucas** [comment]
This does not work for me either. Is there any ways to completely re-build and re-download PostgreSQL once you have already done it the first time?

**Comment by mcandy007** [comment]
Unfortunately, I have the same problem. However, I need postgresql 9.4 for the jsonb support.

Does anyone know of a fully functional "dokku-alt-postgresql:9.4" repo for Dokku-alt?

**Comment by vlucas** [comment]
I also tried doing this from scratch from a blank, new install of dokku-alt. I created a new `/home/dokku/.dokkurc` file to add the line `export POSTGRESQL_IMAGE=ayufan/dokku-alt-postgresql:9.3`, and the started postgres and it still installed 9.1. @ayufan please advise.

**Comment by mcandy007** [comment]
@vlucas The correct Name of the File "/home/dokku/.dokkurc" is without dot. "/home/dokku/dokkurc"

**Comment by vlucas** [comment]
I just followed @ayufan's instructions. I will will try it without the dot tonight when I get a chance.

**Comment by frankwei** [comment]
@vlucas I tried @ayufan's instructions without the dot and it worked.

@mcandy007 I made a copy of 9.4, if you're interested.  I forked @ayufan's 9.3 version and changed everything to 9.4.  I also upgraded the ubuntu version to utopic.  

To use it:
Change the line in `/home/dokku/dokkurc` to `export POSTGRESQL_IMAGE=frankwei/postgresql-dockerfiles`

Then: 
`rm -rf /home/dokku/.pgsql`
`dokku postgresql:restart`

The dockerfile is here: https://registry.hub.docker.com/u/frankwei/postgresql-dockerfiles/dockerfile/

**Comment by mcandy007** [comment]
@frankwei thanks. It works ;-)

**Comment by pcragone** [comment]
Following these instructions has upgraded my postgresql install, but has left my rails app completely unable to connect to the database anymore.  Any thoughts on this?  Any ideas on where I should be looking to debug this issue?
